server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -83,9 +85,37 @@
  */
 @PublicApi(since = "1.0.0")
 public final class IndexSettings {
-    private static final String MERGE_ON_FLUSH_DEFAULT_POLICY = "default";
+    private static final String DEFAULT_POLICY = "default";
     private static final String MERGE_ON_FLUSH_MERGE_POLICY = "merge-on-flush";
 
+    /**
+     * Enum representing supported merge policies
+     */
+    public enum IndexMergePolicy {
+        TIERED("tiered"),
+        LOG_BYTE_SIZE("log_byte_size"),
+        DEFAULT_POLICY(IndexSettings.DEFAULT_POLICY);
+
+        private final String value;
+
+        IndexMergePolicy(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        public static IndexMergePolicy fromString(String text) {
+            for (IndexMergePolicy policy : IndexMergePolicy.values()) {
+                if (policy.value.equals(text)) {
+                    return policy;
+                }
+            }
+            return null;
```

Comment:
I'd suggest throwing an IllegalArgumentException here itself. There should be no reason to return `null` value on invalid inputs.

```suggestion
            throw new IllegalArgumentException(
                "The "
                    + IndexSettings.INDEX_MERGE_POLICY.getKey()
                    + " has unsupported policy specified: "
                    + policy
                    + ". Please use one of: "
                    + String.join(", ", Arrays.stream(IndexMergePolicy.values()).map(IndexMergePolicy::getValue).toArray(String[]::new))
            );
```

Reply:
I was like let the consumer of this enum handle the null case. Given its just this class using `fromString()` I will change it to what you have suggested.